src/app/games/apagos/ApagosRules.ts

```diff
@@ -1,76 +1,112 @@
 import { GameStatus } from 'src/app/jscaip/GameStatus';
 import { GameNode } from 'src/app/jscaip/AI/GameNode';
 import { Player, PlayerOrNone } from 'src/app/jscaip/Player';
-import { Rules } from 'src/app/jscaip/Rules';
+import { ConfigurableRules } from 'src/app/jscaip/Rules';
 import { MGPOptional, MGPValidation, Utils } from '@everyboard/lib';
-import { ApagosCoord } from './ApagosCoord';
 import { ApagosFailure } from './ApagosFailure';
 import { ApagosMove } from './ApagosMove';
 import { ApagosSquare } from './ApagosSquare';
 import { ApagosState } from './ApagosState';
-import { NoConfig } from 'src/app/jscaip/RulesConfigUtil';
 import { PlayerNumberMap } from 'src/app/jscaip/PlayerMap';
+import { NumberConfig, RulesConfigDescription, RulesConfigDescriptionLocalizable } from 'src/app/components/wrapper-components/rules-configuration/RulesConfigDescription';
+import { MGPValidators } from 'src/app/utils/MGPValidator';
+
+export type ApagosConfig = {
+    width: number;
+    increment: number;
+}
 
 export class ApagosNode extends GameNode<ApagosMove, ApagosState> {}
 
-export class ApagosRules extends Rules<ApagosMove, ApagosState> {
+export class ApagosRules extends ConfigurableRules<ApagosMove, ApagosState, ApagosConfig> {
 
-    public static PIECES_PER_PLAYER: number = 10;
 
     private static singleton: MGPOptional<ApagosRules> = MGPOptional.empty();
 
+    // TODO: configurability enhancement 2.0, pour celui qui trouve le ticket: rajouter la possibiliter de restreindre des config multi-paramètres (genre width*increment < X))
+    public static readonly RULES_CONFIG_DESCRIPTION: RulesConfigDescription<ApagosConfig> =
+        new RulesConfigDescription<ApagosConfig>({
+            name: (): string => $localize`Apagos`,
+            config: {
+                width: new NumberConfig(4, RulesConfigDescriptionLocalizable.WIDTH, MGPValidators.range(2, 10)),
```

Comment:
même avec increment de 1, utile d'aller jusqu'à dix :D ?

Reply:
7 it is!